extension/src/popup/components/WarningMessages/index.tsx

```diff
@@ -666,6 +670,138 @@ export const NewAssetWarning = ({
   );
 };
 
+export const UnverifiedTokenWarning = ({
+  domain,
+  code,
+  issuer,
+  onClose,
+}: {
+  domain: string;
+  code: string;
+  issuer: string;
+  onClose: () => void;
+}) => {
+  const { t } = useTranslation();
+  const dispatch: AppDispatch = useDispatch();
+  const warningRef = useRef<HTMLDivElement>(null);
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const publicKey = useSelector(publicKeySelector);
+  const { submitStatus } = useSelector(transactionSubmissionSelector);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const closeOverlay = () => {
+    if (warningRef.current) {
+      warningRef.current.style.marginBottom = `-${POPUP_HEIGHT}px`;
+    }
+    const timeout = setTimeout(() => {
+      onClose();
+      clearTimeout(timeout);
+    }, 300);
+  };
+
+  // animate entry
+  useEffect(() => {
+    if (warningRef.current) {
+      const timeout = setTimeout(() => {
+        // Adding extra check to fix flaky tests
+        if (warningRef.current) {
+          warningRef.current.style.marginBottom = "0";
+        }
+        clearTimeout(timeout);
+      }, 10);
+    }
+  }, [warningRef]);
+
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    await dispatch(
+      addTokenId({
+        publicKey,
+        tokenId: issuer,
+        network: networkDetails.network as Networks,
+      }),
+    );
+    navigateTo(ROUTES.account);
+
+    setIsSubmitting(false);
+  };
+
+  return (
+    <div className="UnverifiedTokenWarning">
+      <View.Content>
+        <div className="UnverifiedTokenWarning__wrapper" ref={warningRef}>
+          <div className="UnverifiedTokenWarning__heading">
+            <div className="UnverifiedTokenWarning__icon">
+              <SorobanTokenIcon code={code} noMargin />
+            </div>
+            <div className="UnverifiedTokenWarning__code">{code}</div>
+            <div className="UnverifiedTokenWarning__domain">{domain}</div>
+            <div className="UnverifiedTokenWarning__description">
+              <div className="UnverifiedTokenWarning__description__icon">
+                <Icon.VerifiedUser />
+              </div>
+              <div className="UnverifiedTokenWarning__description__text">
```

Comment:
Would we show this even when adding a Soroban token? My thought here is that they don't have trustlines so maybe this inly belongs on classic assets or SACs?

Reply:
Yup, we would show this if a user is adding a contract ID that doesn't appear on this list: 
https://api.stellar.expert/explorer/testnet/asset-list/top50

So we just flag that a user is adding a contract ID that has not been vetted

Lmk if that answers your question!

Reply:
Sounds good. I was mainly curious if the badge could be confused with the concept of having a trustline to a classic asset. As in this case we're "trusting" the asset by means of it being on the list which is different than having a trust line on the classic asset side.